Automattic/wp-calypso
# FHT: Add email verification step for business trial plan

Closes https://github.com/Automattic/dotcom-forge/issues/4058

## Proposed Changes
This PR adds an additional step to verify emails when a user email is unverified before they can proceed with the hosting free trial.

This is similar to what's done in the migration free trial, I wanted to re-use that code but opted to create own step instead of trying to refactor migration trial step to match both needs. We might add additional things or change it so I think it's best to create different files.

Also the styling might need some work and it seems from p1696387524534499-slack-C041RHH38NQ that it is not finalise as yet, however I think we can start the discussion with this PR

* Add email verification step
* Show this step only when the business free trial plan is selected

## Testing Instructions



* Have a user with unverified email
* Go to `/setup/new-hosted-site` and select the business free trial plan
* Verify the email verification step is shown
* Resend the email then verify in your email address, 
* Verify email verification step auto refreshes and continues to checkout.

![image](https://github.com/Automattic/wp-calypso/assets/47489215/e2fe50af-5c82-4e5b-9820-aba34ef7a5ba)


## Pre-merge Checklist
